backend/rs/ic_vetkeys/src/utils/mod.rs

```diff
@@ -373,6 +373,91 @@ impl VetKey {
     }
 }
 
+/// Key material derived from a VetKey
+#[derive(Clone, Zeroize, ZeroizeOnDrop)]
+pub struct DerivedKeyMaterial {
+    key: Vec<u8>,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+/// An error while encrypting
+pub enum EncryptionFailed {
+    /// The provided message was too long to be encrypted
+    PlaintextTooLong,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+/// An error while decrypting
+pub enum DecryptionFailed {
+    /// The ciphertext was too short to possibly be valid
+    MessageTooShort,
+    /// The GCM tag did not validate
+    InvalidCiphertext,
+}
+
+impl DerivedKeyMaterial {
```

Comment:
This is mainly a note to myself: One could argue that the intermediate `DerivedKeyMaterial` is only needed in the Typescript variant because there we need it because of the `CryptoKey` wrapper, and that the encryption/decryption functions could live directly on the `VetKey`.

However, having the intermediate `DerivedKeyMaterial` also here is Rust still seems to make sense because (1) then API-wise we are in line with Typescript and (2) it is more apparent/explicit that the encryption doesn't happen with the vetKey itself but with a key that is derived from it.

Reply:
Exactly so, this could have easily also been some freestanding functions, but having the functions have similar names/interfaces between the two libraries seemed like it would make a life a bit easier for developers